### ISSUE TITLE
Lemmatization results not consistent across different python sessions

### ISSUE THREAD

**Issue by AnastasisB** [explosion/spaCy#9607]
## How to reproduce the behaviour

```
import spacy
nlp = spacy.load("el_core_news_lg")
doc = nlp("καλησπερες")

for token in doc:
    print(token.lemma_)
```

I came across #3484 where there was a discussion about inconsistency between the results of lemmatization. Seems like the problem still persists. [In this line](https://github.com/explosion/spaCy/blob/657af5f91f88bb5a414ae133a99465dbc4f240be/spacy/pipeline/lemmatizer.py#L224)   the list is created (not sorted), and [in this line](https://github.com/explosion/spaCy/blob/657af5f91f88bb5a414ae133a99465dbc4f240be/spacy/pipeline/lemmatizer.py#L111), first element of the list is used as result/lemma (please correct me if I'm wrong).

Even If I'm using single words (like in the example above), the result is not consistent and it varies between the elements of the list (eg. ['καλησπερα', 'καλησπερε']).
As a workaround I sorted the list before getting it's first element [in line 111](https://github.com/explosion/spaCy/blob/657af5f91f88bb5a414ae133a99465dbc4f240be/spacy/pipeline/lemmatizer.py#L111), and it seems to be working as intended.

```
token.lemma_ = sorted(self.lemmatize(token))[0]
```

Am I missing something here?

## Your Environment

- **spaCy version:** 3.0.0
- **Platform:** Windows-10-10.0.19041-SP0
- **Python version:** 3.7.4
- **Pipelines:** el_core_news_lg (3.0.0), el_core_news_sm (3.0.0)

**Comment by polm** [comment]
This has cropped up a couple of times, and was most recently in fixed in #9405, a little under a month ago, though that didn't involve Greek. I can't reproduce your issue in the latest 3.1.4 so I suspect it was fixed before then. Do you still have the issue if you upgrade from 3.0.0?

**Comment by AnastasisB** [comment]
Even though I'm having a totally different lemma this time (that wasn't existing in the previous list), the result seems to be consistent. Issue seems to be resolved. Thank you for the response.
